zerver/lib/retention.py

```diff
@@ -585,3 +586,12 @@ def clean_archived_data() -> None:
         count += len(transaction_block)
 
     logger.info("Deleted %s old ArchiveTransactions.", count)
+
+def parse_message_retention_days(value: Union[int, str],
+                                 special_values_map: Dict[str, Optional[int]]) -> Optional[int]:
+    if isinstance(value, str) and value in special_values_map.keys():
```

Comment:
@shubhamdhama that mypy error is fixed if I specify `isinstance` but fails when written this way,
```js
    if value in special_values_map.keys():
```